pkg/plugins/manager/loader/loader.go

```diff
@@ -312,28 +335,65 @@ func (l *Loader) readPluginJSON(pluginJSONPath string) (plugins.JSONData, error)
 	return plugin, nil
 }
 
-func createPluginBase(pluginJSON plugins.JSONData, class plugins.Class, pluginDir string) *plugins.Plugin {
+func createPluginBase(pluginJSON plugins.JSONData, class plugins.Class, pluginDir string, isCDN bool, cdnBasePath string) (*plugins.Plugin, error) {
```

Comment:
Many thanks for the review!
I agree it was a bit confusing before, I have refactored it now so it uses two services:

- `pluginscdn.Service` which contains all CDN related stuff (like `IsEnabled()`, so we avoid direct comparisons against the plugin settings or the plugins cdn section to see if CDN is enabled)
- `assetpath.Service` which can construct base, module.js and relative paths for assets. It depends on `plugisncdn.Service` and it will return a local or CDN path depending on the plugin and on the instance's config

I split it into two services rather than putting the path constructing methods into `pluginscdn.Service` mainly for naming reasons (`moduleURL, err := l.cdn.Module(...)` wouldn't make sense since it could return a non-cdn path as well), and because local path construction should not be a concern of the cdn service anyways.

What do you think now? I have also left some comments because I believe we should also move the fallback redirect handler and its prometheus metric into the `plugisncdn.Service` service to keep everything in one place, but I am not sure if having handlers into separate services is encouraged in core or not (all the plugin API handlers seem to be into the `api.HTTPServer` service)